pkg/utils/dataset/lifecycle/node.go

```diff
@@ -258,3 +258,22 @@ func labelNodeWithCapacityInfo(toUpdate *v1.Node, runtimeInfo base.RuntimeInfoIn
 	toUpdate.Labels[totalCapacityLabel] = totalValue
 	*updatedLabels = append(*updatedLabels, totalCapacityLabel)
 }
+
+// UpdateOrDeleteDatasetNum deletes the datasetNum label or updates the number of the dataset in the specific node.
+func UpdateOrDeleteDatasetNum(toUpdate *v1.Node, runtimeInfo base.RuntimeInfoInterface) (isDeleted bool, err error) {
```

Comment:
Could you add test for this function? No need to do it today.

Reply:
Okay, I will pay more attention to it later.